test/acceptance/features/bindAppToServiceAnnotations.feature

```diff
@@ -542,3 +542,49 @@ Feature: Bind an application to a service using annotations
             """
         Then Service Binding "binding-app-via-gvk" is ready
         And The application env var "BACKEND_HOST_INTERNAL_DB" has value "internal.db.stable.example.com"
+
+    Scenario: Bind application to service with invalid sourceValue annotations
+        Given Generic test application is running
+        And CustomResourceDefinition backends.stable.example.com is available
+        And The Custom Resource is present
+            """
+            apiVersion: stable.example.com/v1
+            kind: Backend
+            metadata:
+                name: $scenario_id
+                annotations:
+                    service.binding/webarrows: path={.spec.connections},elementType=sliceOfMaps,sourceKey=type,sourceValue=asdf
+            spec:
+                connections:
+                  - type: primary
+                    url: primary.example.com
+                  - type: secondary
+                    url: secondary.example.com
+                  - type: '404'
+                    url: black-hole.example.com
+            status:
+                somestatus: good
+            """
+        When Service Binding is applied
+            """
+            apiVersion: binding.operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: $scenario_id
+            spec:
+                bindAsFiles: false
+                services:
+                  - group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: $scenario_id
+                application:
+                    name: $scenario_id
+                    group: apps
+                    version: v1
+                    resource: deployments
+            """
+        Then Service Binding CollectionReady.status is "False"
+	And Service Binding CollectionReady.reason is "ValueNotFound"
+	And Service Binding CollectionReady.message is "Value for key webarrows_primary not found"
```

Comment:
acceptance tests are failing on CI on this assertion, please double check what is going on.

Reply:
I've looked into [the artifacts of the failing without-olm job](https://github.com/redhat-developer/service-binding-operator/suites/3715396067/artifacts/90835111) and found that this step is failing because the message is actually `Value for key webarrows_secondary not found`:
```shell
$ cat servicebindings.binding.operators.coreos.com.yaml| yq -r '.items[] | select(.metadata.name == "bindapptoserviceannotations-546").status.conditions[] | select(.type == "CollectionReady")'

{
  "lastTransitionTime": "2021-09-08T17:15:27Z",
  "message": "Value for key webarrows_secondary not found",
  "reason": "ValueNotFound",
  "status": "False",
  "type": "CollectionReady"
}
```

For comparison by checking the [artifacts of the passed with-olm job](https://github.com/redhat-developer/service-binding-operator/suites/3715396067/artifacts/90835110) the message is correct:
```shell
$ cat servicebindings.binding.operators.coreos.com.yaml| yq -r '.items[] | select(.metadata.name == "bindapptoserviceannotations-546").status.conditions[] | select(.type == "CollectionReady")'
{
  "lastTransitionTime": "2021-09-08T17:15:50Z",
  "message": "Value for key webarrows_primary not found",
  "reason": "ValueNotFound",
  "status": "False",
  "type": "CollectionReady"
}
```

Reply:
It looks like whatever is going through `.spec.connections` map keys do it in a different order for each of the above cases